Sources/Datadog/DatadogInternal/Context/DatadogContext.swift

```diff
@@ -57,20 +57,24 @@ internal struct DatadogContext {
     let device: DeviceInfo
 
     /// Current user information.
-    var userInfo: UserInfo
+    var userInfo: UserInfo?
 
-    /// Application launch time..
-    var launchTime: LaunchTime
+    /// Application launch time.
+    ///
+    /// Can be `nil` if the launch could not yet been evaluated.
+    var launchTime: LaunchTime?
 
     /// Provides the history of app foreground / background states.
-    var applicationStateHistory: AppStateHistory
+    var applicationStateHistory: AppStateHistory?
 
     // MARK: - Device Specific
 
     /// Network information.
     ///
     /// Represents the current state of the device network connectivity and interface.
-    var networkConnectionInfo: NetworkConnectionInfo
+    /// The value can be `nil` if the network interface is not available or if it has not
+    /// yet been evaluated.
+    var networkConnectionInfo: NetworkConnectionInfo?
```

Comment:
Conditional init of the `DatadogContextProvider` prevents from keeping those as non-optional :( even tho they all be set to their respective publisher `initialValue`.

Reply:
Yes, that's true! let me rethink the init flow.